homeassistant/components/lg_thinq/water_heater.py

```diff
@@ -0,0 +1,211 @@
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import ThinqConfigEntry
+from .coordinator import DeviceDataUpdateCoordinator
+from .entity import ThinQEntity
+
+DEVICE_TYPE_WH_MAP: dict[DeviceType, WaterHeaterEntityDescription] = {
+    DeviceType.WATER_HEATER: WaterHeaterEntityDescription(
+        key=ExtendedProperty.WATER_HEATER,
+        name=None,
+        translation_key=ExtendedProperty.WATER_HEATER,
+    ),
+    DeviceType.SYSTEM_BOILER: WaterHeaterEntityDescription(
+        key=ExtendedProperty.WATER_BOILER,
+        translation_key=ExtendedProperty.WATER_BOILER,
+    ),
+}
+
+# Mapping between device and HA operation modes
+OP_STATE_TO_HA = {
+    "auto": STATE_ECO,  # Auto
+    "heat_pump": STATE_HEAT_PUMP,  # Heat pump
+    "turbo": STATE_PERFORMANCE,  # Turbo
+    "vacation": STATE_OFF,  # Vacation
+}
+HA_STATE_TO_OP = {v: k for k, v in OP_STATE_TO_HA.items()}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ThinqConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up an entry for water_heater platform."""
+    entities: list[ThinQWaterHeaterEntity] = []
+    for coordinator in entry.runtime_data.coordinators.values():
+        if (
+            description := DEVICE_TYPE_WH_MAP.get(coordinator.api.device.device_type)
+        ) is not None:
+            if coordinator.api.device.device_type == DeviceType.WATER_HEATER:
+                entities.extend(
+                    [
+                        ThinQWaterHeaterEntity(
+                            coordinator, description, ExtendedProperty.WATER_HEATER
+                        )
+                    ]
+                )
+            elif coordinator.api.device.device_type == DeviceType.SYSTEM_BOILER:
+                entities.extend(
+                    [
+                        ThinQWaterBoilerEntity(
+                            coordinator, description, ExtendedProperty.WATER_BOILER
+                        )
+                    ]
+                )
+    if entities:
+        async_add_entities(entities)
+
+
+class ThinQWaterHeaterEntity(ThinQEntity, WaterHeaterEntity):
+    """Represent a thinq water_heater platform."""
+
+    entity_description: WaterHeaterEntityDescription
+
+    def __init__(
+        self,
+        coordinator: DeviceDataUpdateCoordinator,
+        entity_description: WaterHeaterEntityDescription,
+        property_id: str,
+    ) -> None:
+        """Initialize a water_heater entity."""
+        super().__init__(coordinator, entity_description, property_id)
+        self._attr_supported_features = (
+            WaterHeaterEntityFeature.TARGET_TEMPERATURE
+            | WaterHeaterEntityFeature.OPERATION_MODE
+        )
+        self._attr_temperature_unit = (
+            self.coordinator.hass.config.units.temperature_unit
+        )
```

Comment:
Why do we do this if we overwrite it in the `_update_status`?

Reply:
This is still unresolved too